projects/plugins/jetpack/extensions/plugins/publicize/components/panel/index.js

```diff
@@ -19,21 +19,30 @@ import PublicizeConnectionVerify from '../connection-verify';
 import PublicizeForm from '../form';
 import PublicizeTwitterOptions from '../twitter/options';
 import useSelectSocialMediaConnections from '../../hooks/use-social-media-connections';
-import usePostJustSave from '../../hooks/use-post-just-saved';
+import { usePostJustBeforeToPublish } from '../../hooks/use-saving-post';
 
 const PublicizePanel = ( { prePublish } ) => {
-	const { refresh, hasEnabledConnections } = useSelectSocialMediaConnections();
+	const { refresh, connections } = useSelectSocialMediaConnections();
 
 	// Refresh connections when the post is just saved.
-	usePostJustSave(
+	usePostJustBeforeToPublish(
 		function () {
-			if ( ! hasEnabledConnections ) {
-				return;
-			}
-
-			refresh();
+			/*
+			 * Being optimistic, it sets the connections
+			 * that are going to be used
+			 * to share the post as `done`.
+			 * The sharing process is handled by an async action,
+			 * in the server-side.
+			 */
+			const updatedConnections = connections.map( connection => ( {
+				...connection,
+				done: connection.enabled,
+				toggleable: ! connection.enabled,
```

Comment:
I think we'll want `toggleable` to be set based on whether Republicize is available

Reply:
yes, but Republicize is a feature that we even have not started to work on, correct?

Reply:
yes, agree. I just want to confirm that's something we shouldn't take care of for now.